torch/utils/data/datapipes/iter/grouping.py

```diff
@@ -179,7 +181,8 @@ def __iter__(self) -> Iterator[List[T_co]]:
         else:
             bucket: List[T_co]
             batch: List[T_co] = []
-            for bucket in self.bucket_ds:
+            for bucket_or_chunk in self.bucket_ds:
+                bucket = list(bucket_or_chunk)
                 # In-place sort within bucket
                 bucket.sort(key=self.sort_key)
```

Comment:
If `DataChunk` inherits from `list` then no need to convert to list and run in-place sort. And, then converting list back to `DataChunk` is not needed.

Reply:
This is weird reason, but I would rather keep explicit conversion here, because it looks ugly and will convert it sooner in this case (while adding support for DF)